src/components/MTableBody/index.js

```diff
@@ -0,0 +1,353 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TableBody, TableCell, TableRow } from '@material-ui/core';
+
+function MTableBody({
+  columns,
+  options,
+  pageSize,
+  localization,
+  icons,
+  actions,
+  detailPanel,
+  hasDetailPanel,
+  onEditingApproved,
+  onEditingCanceled,
+  onBulkEditRowChanged,
+  getFieldValue,
+  scrollWidth,
+  components,
+  errorState,
+  isTreeData,
+  bulkEditOpen,
+  onRowSelected,
+  currentPage,
+  onToggleDetailPanel,
+  onRowClick,
+  onTreeExpandChanged,
+  onFilterChanged,
+  hasAnyEditingRow,
+  treeDataMaxLevel,
+  cellEditable,
+  onCellEditStarted,
+  onCellEditFinished,
+  onGroupExpandChanged,
+  showAddRow,
+  renderData,
+  initialFormData
+}) {
+  const renderDataC = renderData;
```

Comment:
What is the purpose of this copy here?

Reply:
there was a naming conflict so I had to rename the function? But I do agree that there is more work that could be done

Reply:
You can simply rename it in the destructure which makes it clearer {oldName: newName} to avoid name conflicts.